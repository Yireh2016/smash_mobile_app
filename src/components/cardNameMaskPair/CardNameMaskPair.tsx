import * as React from 'react';
import { StyleSheet, View } from 'react-native';
import { NewCard } from '../../types/Card';
import { P1 } from '../texts/Texts';
import truncateText from '../../helpers/truncateText/truncateText';

interface CardNameMaskPairProps {
  card: NewCard;
  maxLentgh?: number;
}
const styles = StyleSheet.create({
  container: {
    flexDirection: 'row',
  },
});
const CardNameMaskPair: React.FunctionComponent<CardNameMaskPairProps> = ({
  card,
  maxLentgh = 16,
}) => {
  return (
    <View style={styles.container}>
      <P1 isBold={true}>{`${truncateText(card.bank_name, maxLentgh)}`}</P1>
      <P1>{` (${card.mask})`}</P1>
    </View>
  );
};

export default CardNameMaskPair;
