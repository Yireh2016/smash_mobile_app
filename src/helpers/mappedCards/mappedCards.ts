import { AprPair, Card } from '../../types/Card';

export default (cards: Card[]) =>
  cards.map((card: Card) => {
    const {
      account_id,
      bank,
      number,
      current_balance,
      credit_utilization_percentage,
      statement_balance,
      purchase_apr_percentage,
      other_aprs,
      min_payment,
      payment_due,
    } = card;

    const aprs = other_aprs.map((apr: AprPair) => {
      const { balance_subject_to_apr, apr_percentage, apr_type } = apr;
      return {
        apr_percentage: apr_percentage,
        balance_subject_to_apr: balance_subject_to_apr,
        apr_type,
      };
    });
    return {
      account_id,
      bank_name: bank,
      mask: number,
      current_balance,
      credit_utilization_percentage,
      purchase_apr_percentage,
      aprs,
      last_statement_balance: statement_balance,
      minimum_payment_amount: min_payment,
      next_payment_due_date: payment_due,
    };
  });
