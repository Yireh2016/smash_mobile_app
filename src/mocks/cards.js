export default [
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '2267',
    current_balance: 500,
    credit_utilization_percentage: 10,
    statement_balance: 2650,
    purchase_apr_percentage: 19,
    min_payment: 65,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 19,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 3100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9376',
    current_balance: 2000,
    credit_utilization_percentage: 25,
    statement_balance: 1030,
    purchase_apr_percentage: 21,
    min_payment: 37,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 21,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1100,
      },
    ],
  },
  {
    account_id: '2u3ywjhedb',
    bank: 'Citi',
    number: '9085',
    current_balance: 3000,
    credit_utilization_percentage: 10,
    statement_balance: 1820,
    purchase_apr_percentage: 22,
    min_payment: 42,
    payment_due: '2020-12-18',
    other_aprs: [
      {
        apr_type: 'purchase_apr',
        apr_percentage: 22,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'cash_apr',
        apr_percentage: 27.95,
        balance_subject_to_apr: 0,
      },
      {
        apr_type: 'adjustment',
        apr_percentage: 0,
        balance_subject_to_apr: 1900,
      },
    ],
  },
];
