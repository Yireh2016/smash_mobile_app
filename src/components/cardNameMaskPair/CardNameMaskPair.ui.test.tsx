import * as React from 'react';
import renderer from 'react-test-renderer';
import mappedCards from '../../helpers/mappedCards/mappedCards';
import cards from '../../mocks/cards';
import CardNameMaskPair from './CardNameMaskPair';

it('CardNameMaskPair renders correctly', () => {
  const Element = () => {
    return <CardNameMaskPair card={mappedCards([cards[0]])[0]} />;
  };

  const rendered = renderer.create(<Element />).toJSON();
  expect(rendered).toMatchSnapshot();
});
