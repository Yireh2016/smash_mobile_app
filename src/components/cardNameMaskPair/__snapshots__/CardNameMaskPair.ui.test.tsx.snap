// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardNameMaskPair renders correctly 1`] = `
<View
  style={
    Object {
      "flexDirection": "row",
    }
  }
>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "OpenSans-Bold",
        "fontSize": 16,
      }
    }
  >
    Citi
  </Text>
  <Text
    style={
      Object {
        "color": "#fff",
        "fontFamily": "OpenSans-Regular",
        "fontSize": 16,
      }
    }
  >
     (2267)
  </Text>
</View>
`;
